staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest_test.go

```diff
@@ -173,32 +173,46 @@ func (p *testPatcher) New() runtime.Object {
 }
 
 func (p *testPatcher) Update(ctx context.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc) (runtime.Object, bool, error) {
-	currentPod := p.startingPod
-	if p.numUpdates > 0 {
-		currentPod = p.updatePod
-	}
-	p.numUpdates++
+	// Simulate GuaranteedUpdate behavior (retries internally on etcd changes if the incoming resource doesn't pin resourceVersion)
+	for {
+		currentPod := p.startingPod
```

Comment:
I'm having trouble understanding this-- why is there no locking? do we not have a GuaranteedUpdate fake anywhere? Why does the patcher have startingPod and updatePod as member variables?

The diff might be more comprehensible if you move the body of the loop into its own function.

Reply:
> why is there no locking?

the test is single-threaded

> do we not have a GuaranteedUpdate fake anywhere?

we do not :(

> Why does the patcher have startingPod and updatePod as member variables?

it only models a single conflict/retry scenario

> The diff might be more comprehensible if you move the body of the loop into its own function.

I can if you'd like, or you can view the commit ignoring whitespace changes https://github.com/kubernetes/kubernetes/commit/9f6d91f4c979fdc30067c7f3e7eabb8401da99c9?w=1